### PR TITLE
[maistra-2.2] OSSM-4950: Fix CVE CVE-2023-44487

### PR DESCRIPTION
**Pull request by tedjpoole**
Replaces #296

### CODE REVIEW

File: source/common/http/conn_manager_impl.cc [hunk 1]
```diff
@@ -1215,8 +1241,11 @@ void ConnectionManagerImpl::ActiveStream::decodeTrailers(RequestTrailerMapPtr&&
 
   ASSERT(!request_trailers_);
   request_trailers_ = std::move(trailers);
-  filter_manager_.maybeEndDecode(true);
-  filter_manager_.decodeTrailers(*request_trailers_);
+  filter_manager_.maybeEndDecode(true); // TODO(ted): Tim's fix did this
+  // maybeEndDecode(true);              // TODO(ted): Upstream's fix did this
```
Comment (tedjpoole) [inline source/common/http/conn_manager_impl.cc]: I didn't drag in the `ConnectionManagerImpl::ActiveStream::maybeEndDecode()` method yet. I wasn't sure what it added beyond just calling `filter_manager_.maybeEndDecode()`, so still need to look into it to answer conclusively.

**Comment by twghu** [comment]
Rechecked commit changes and ran tests locally and all passed.

**Review by twghu — APPROVED** [review]

**Pull request merged**